vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/codec/json/JsonObjectCodec.java

```diff
@@ -106,6 +149,14 @@ protected Object readObjectId(BsonReader reader, DecoderContext ctx) {
 
   @Override
   protected Object readDateTime(BsonReader reader, DecoderContext ctx) {
```

Comment:
OK, here's what the Java parser accepts for dates in JSON (I can post a gist if that helps):

```
{"myDateField": {"$date": <epochMillis>}} -- in this case epcohMills must be an INT32 or INT64 (see visitDateTimeExtendedJson)

{"myDateField": Date("<dateString>")} -- in this case dateString must have the format EEE MMM dd yyyy HH:mm:ss z (see visitDateTimeConstructor)
{"myDateField": Date(epochMillis)} -- in this case epochMillis must be a INT32 or INT64 value

{"myDateField": ISODate("<dateString>")} -- in this case dateString must have one of the following formats: "yyyy-MM-dd", "yyyy-MM-dd'T'hh:mm:ssz", "yyyy-MM-dd'T'hh:mm:ss.SSSz"
```

So... the first format is what @zigzago implemented to start with (I think).  The others allow for some variant of the "shell" mode to appear in JSON strings and be correctly parsed.  Note that none of these exactly match the strict mode implemented by the Mongo parser itself (this is part of the reason why date handling in Mongo has always driven me insane).

Probably the first one is our best bet for something that will pass through both the JsonObject parser and the org.bson.Document parser (which would be a good thing IMO).